File: src/com/heiwawu/example/leason1/AnimalType.java
package com.heiwawu.example.leason1;

public enum AnimalType {
    CAT(1,"猫"),
    DOG(2, "狗");
    private int value;
    private String displayName;

    private AnimalType(int value, String displayName){
        this.value = value;
        this.displayName = displayName;
    }

    /**
     * 获取枚举的 int 值,用于数据保存以及序列化
     *
     * @return 枚举的 int 值
     */
    public int value() {
        return this.value;
    }

    /**
     * 获取枚举的显示名称
     *
     * @return 枚举的显示名称
     */
    public String displayName() {
        return this.displayName;
    }

}

File: gosrc/leason1/animaltype.go
package leason1

type AnimalEnumType int
type Func func() (string, error)

const (
	_ AnimalEnumType = iota * 50
	CAT
	DOG
)

func NewFunc() {
	var fn Func = func() (string, error) {
		return "", nil
	}
	fn()
}
